crates/compiler/src/visitors/string_table_generator_visitor.rs

```diff
@@ -0,0 +1,236 @@
+use crate::compiler;
+use crate::prelude::generated::{yarnspinnerparser::*, yarnspinnerparservisitor::*};
+use crate::prelude::*;
+use antlr_rust::parser_rule_context::ParserRuleContext;
+use antlr_rust::token::Token;
+use antlr_rust::tree::{ParseTree, ParseTreeVisitorCompat, Tree};
+use std::fmt::Debug;
+use std::rc::Rc;
+
+#[derive(Debug, Clone)]
+/// A Visitor that walks an expression parse tree and generates string
+/// table entries, which are provided to a [`StringTableManager`].
+/// This string table can then be provided
+/// to future compilation passes, or stored for later use. Call the
+/// [`visit`] method to begin generating string table entries.
+pub(crate) struct StringTableGeneratorVisitor {
+    pub(crate) diagnostics: Vec<Diagnostic>,
+    current_node_name: String,
+    file_name: String,
+    pub(crate) string_table_manager: StringTableManager,
+}
+
+impl StringTableGeneratorVisitor {
+    pub(crate) fn new(file_name: String, string_table_manager: StringTableManager) -> Self {
+        Self {
+            file_name,
+            string_table_manager,
+            diagnostics: Default::default(),
+            current_node_name: Default::default(),
+        }
+    }
+}
+
+impl ParseTreeVisitorCompat<'_> for StringTableGeneratorVisitor {
+    type Node = YarnSpinnerParserContextType;
+
+    type Return = ();
+
+    fn temp_result(&mut self) -> &mut Self::Return {
+        Box::leak(Box::new(()))
+    }
+}
+
+impl<'input> YarnSpinnerParserVisitorCompat<'input> for StringTableGeneratorVisitor {
+    fn visit_node(&mut self, ctx: &NodeContext<'input>) -> Self::Return {
+        let mut tags = Vec::new();
+        for header in ctx.header_all() {
+            let header_key = header.header_key.as_ref().unwrap().get_text();
+            if header_key == "title" {
+                self.current_node_name =
+                    header.header_value.as_ref().unwrap().get_text().to_owned();
+            } else if header_key == "tags" {
+                let header_value = header
+                    .header_value
+                    .as_ref()
+                    .map(|header| header.get_text())
+                    .unwrap_or_default();
+                // Split the list of tags by spaces, and use that
+                // Todo: This seems to always take the last tags, idk if that is intended
+                // Todo: Original splits by spaces exclusively, this splits by any whitespace -> Check if this is okay
+                tags = header_value
+                    .split_whitespace()
+                    .map(ToOwned::to_owned)
+                    .collect();
+            }
+        }
+        if !self.current_node_name.is_empty() && tags.contains(&"rawText".to_owned()) {
+            // This is a raw text node. Use its entire contents as a
+            // string and don't use its contents.
```

Comment:
> Use its entire contents as a string and don't use its contents.

What's the difference?

Reply:
idk, I copied this comment. Sounds weird tho 🤷